### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by sammyboy2000**

### CODE REVIEW

**Comment by Ashrom0** [comment]
I will review this.

**Review by Ashrom0 — COMMENTED** [review]
Create two interfaces. One to represent the JokeService and one interface to represent outputting the joke to the screen (Console.WriteLine). ✔
The JokeService will need to have the interface applied to it. ❌
Create an implementation of the output interface that writes out the joke to the console. ✔
Implement the Jester class. It should take in both interfaces as dependencies. These dependencies should be null checked. ✔
The Jester class TellJoke() method should retrieve a joke from the JokeService. If the joke contains "Chuck Norris", skip it and get another. The joke should be written to the output dependency. ❌
Unit test the Jester class. Code coverage should be above 90% for this class. Moq will make this significantly easier. ✔
Enable nullability in the .csproj files and ensure no errors or warnings: ✔
Ensure you enable:
nullable reference types is enabled ✔
net6 targetted ✔
C# 10.0 ✔
and enabled .NET analyzers for both projects ✔
For this assignment, always use Assert.AreEqual<T>() (the generic version) ✔
All of the above should be unit tested ✔
Choose simplicity over complexity ✔

**Comment by Ashrom0** [comment]
Although the program seems to run good. I see that the JokeService was taken out and replaced for another method and since its missing, I dont see the telljoke method that retrieves from jokeservices, but you still were able to run the program just fine.

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Net.Http;
+
+namespace CanHazFunny
+{
+    public class Jester : IOutput
+    {
+        public HttpClient HttpClient { get; }
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:8]: This should be the IOutput class.

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        public JokeService JokeService { get; }
+        public Jester()
+        {
+            HttpClient = new HttpClient();
+            JokeService = new JokeService();
+        }
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:14]:
```suggestion
        public Jester(IJokeService jokeService, IOutputService outputService)
        {
            OutputService = outputService ?? null check here;
            JokeService = jokeService ?? null check here;
        }
```

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            string joke;
+            do
+            {
+                joke = JokeService.GetJoke(HttpClient);
+            } while (!ScreenJoke(joke, filters));
+
+            IOutput.PresentJoke(joke);
+        }
+
+        public static bool ScreenJoke(string joke, string[] filters)
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:27]: Maybe make this private?

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            if (joke == null || filters == null || filters.Length == 0)
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:29]: Good idea making the filters a passed-in a variable. It makes your method a lot more flexible and easy to change/refactor in the future.

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 5]
```diff
@@ -1,10 +1,60 @@
...
+using System;
+using System.IO;
 
 namespace CanHazFunny.Tests
 {
     [TestClass]
     public class JesterTests
     {
-        
+        [TestMethod]
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:10]: You could benefit from more tests here, including tests for the JokeService class.

**Pull request closed without merging**